pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -90,9 +90,35 @@ type ArtifactGCStrategy string
 const (
 	ArtifactGCOnWorkflowCompletion ArtifactGCStrategy = "OnWorkflowCompletion"
 	ArtifactGCOnWorkflowDeletion   ArtifactGCStrategy = "OnWorkflowDeletion"
+	ArtifactGCOnWorkflowSuccess    ArtifactGCStrategy = "OnWorkflowSuccess"
```

Comment:
do we need these, I'm the problem with them is that they'll result in orphan artifacts that never get GCd, I think mostly users should use "OnDeletion", so I'd like to see clear customer need for this

Reply:
which do you mean? "OnWorkflowSuccess" and "OnWorkflowFailure"? In the Community meeting when I presented people were saying that if their Workflow fails they want to be able to have the Artifacts to understand what happened, if I'm recalling correctly. Or maybe you want to only preserve the artifacts if everything worked correctly. We can get clarification from Bala/Terry if you want.